### PR TITLE
feat(ai): taxonomy queries

### PR DESCRIPTION
**Pull request by skoob13**
## Problem

We have EventDefinition and PropertyDefinition models, but they don't provide essential fields for the assistant (volume for sorting), and there aren't relationships between models. To optimize taxonomy discovery, we want to reduce the noise as much as possible by excluding stale events and their properties, e.g. providing a narrowed list of properties for a specific event.

This PR adds two new queries:
- Events for the last 30d with their volume. The event discovery agent will use this query to build a first prompt.
- Properties, example property values, and the count of distinct properties for an event for the last 30d. The event discovery agent will use this query to discover properties and choose values for filtering.

Additionally, I plan to add another query or extend the existing one for smaller orgs to retrieve the taxonomy using a single query instead of 1 + N queries. It's not a viable solution for larger orgs since we can't do a query in ClickHouse on scale (either timeout exception or OOMs).

## Changes

Adds two queries and tests for them.

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Unit tests, manual testing.

### CODE REVIEW

**Review by aspicer — APPROVED** [review]
Looks great! 

Seems super useful, don't know how long you plan on caching these for or how often you'd want to refresh them, but maybe we could optimize these queries a bit down the road with sampling, or make sure we're caching them adequately.

File: frontend/src/queries/schema.ts [hunk 1]
```diff
@@ -1979,3 +1983,31 @@ export interface TrendsAlertConfig {
...
+    sample_values: string[]
+    sample_count: integer
+}
+
+export interface EventTaxonomyQuery extends DataNode<EventTaxonomyQueryResponse> {
+    kind: NodeKind.EventTaxonomyQuery
+    event: string
+}
+
+export type EventTaxonomyQueryResponse = AnalyticsQueryResponseBase<EventTaxonomyResponse[]>
```
Comment (Twixes) [inline frontend/src/queries/schema.ts:2011]:
```suggestion
export type EventTaxonomyResponse = {
    property: string
    sample_values: string[]
    sample_count: integer
}[]

export interface EventTaxonomyQuery extends DataNode<EventTaxonomyQueryResponse> {
    kind: NodeKind.EventTaxonomyQuery
    event: string
}

export type EventTaxonomyQueryResponse = AnalyticsQueryResponseBase<EventTaxonomyResponse>
```

File: frontend/src/queries/schema.ts [hunk 2]
```diff
@@ -1979,3 +1983,31 @@ export interface TrendsAlertConfig {
 export interface HogCompileResponse {
     bytecode: any[]
 }
+
+export interface TeamTaxonomyResponse {
```
Comment (Twixes) [inline frontend/src/queries/schema.ts:1987]: `...Response` naming only applies to the final whole response, while this is a subpart of it. Something more specific would be clearer here, e.g. `EventInfo`

**Review by Twixes — APPROVED** [review]
LGTM, although with notes on the naming of a couple types

**Pull request merged**